lib/core.js

```diff
@@ -295,18 +295,26 @@ declare class JSON {
 }
 
 /* Iterators */
-interface IteratorResult<T> {
+interface IteratorResult<Y,R> {
```

Comment:
Just a bit of future proofing. Once #577 is fixed, we can use a better type for this that takes advantage of this parameterization:

`type IteratorResult = { done: false, value: Y } | { done: true, value: ?R }`

(Annoyingly, the `value` member is `R` the *first time* `done` switches to `true`, but if you call `next` again, it becomes `undefined`. >:(.